src/plot.js

```diff
@@ -592,8 +602,7 @@ function maybeTopFacet(facet, options) {
   if (facet == null) return;
   const {x, y} = facet;
   if (x == null && y == null) return;
-  const data = arrayify(facet.data);
-  if (data == null) throw new Error(`missing facet data`);
```

Comment:
It's unexpected that we'd remove this error message (in 85d81ab486). Same in maybeMarkFacet.

Reply:
This is needed or else Plot.raster/Plot.contour will crash when faceted when used in conjunction with the sampler initializer because the data is null.

Reply:
Latest commit restores the error when `data` is strictly undefined which should catch the case where the user has not specified the data (or specified it incorrectly _e.g._ `facet: {data2}`), but if it’s set explicitly to null as in the case of the raster/contour mark, it’s allowed.